src/clj/puppetlabs/puppetserver/ringutils.clj

```diff
@@ -85,6 +109,28 @@
         false))
     true))
 
+(schema/defn ^:always-validate request->name :- (schema/maybe schema/Str)
+  "Return the name of the client by parsing the subject of the client cert from
+  the request.  Return nil if the client cert is absent or cannot be parsed."
+  [request :- RingRequest]
+  (let [{:keys [ssl-client-cert]} request]
+    (if ssl-client-cert (get-cert-subject ssl-client-cert))))
+
+(schema/defn ^:always-validate authorized? :- schema/Bool
+  "(SERVER-768) replacement for client-allowed-access?"
+  [rules :- rules/Rules
+   request :- RingRequest]
+  (if-let [name (request->name request)]
```

Comment:
I noticed that the library has its own middleware function, `wrap-authorization-check`, which derives the `name` a little bit differently.  It pulls it from the SSL client certificate, if available, else falls back to a reverse lookup on the `:remote-addr` -- https://github.com/masterzen/trapperkeeper-authorization/blob/master/src/puppetlabs/trapperkeeper/authorization/ring_middleware.clj

Presumably, this is more like how this was done with legacy auth.conf handling.  If we were using this middleware instead of what is being done here, I think my comment here would be irrelevant.  Is there a reason you didn't go with this approach?